src/dht/dht-peer-routing.js

```diff
@@ -26,14 +27,53 @@ class DHTPeerRouting {
    * @param {any} options
    */
   async findPeer (peerId, options = {}) {
+    /** @type {PeerData} */
+    const result = {
+      id: peerId,
+      multiaddrs: []
+    }
+    let peerFound = false
     for await (const event of this._dht.findPeer(peerId, options)) {
-      if (event.name === 'PEER_RESPONSE') {
-        const peer = event.closer.find(peerData => peerData.id.equals(peerId))
+      let multiaddrs 
+      switch (event.name) {
+        case 'FINAL_PEER':
+          peerFound = true
+          multiaddrs = event.peer.multiaddrs
+          break
+        case 'PEER_RESPONSE':
+          peerFound = true
```

Comment:
`PEER_RESPONSE` just means a network peer responded to a query, it doesn't necessarily mean the peer we're looking for will be in the `closer` field of the response - it could just contain peers that are closer to the peer we are looking for.

Reply:
Oh, in this case, there are some other unit tests wrong - because they don't assume the `FINAL_PEER` event